firefox-ios/Client/Frontend/Browser/MainMenuActionHelper.swift

```diff
@@ -720,7 +722,8 @@ class MainMenuActionHelper: PhotonActionSheetProtocol,
     }
 
     private func getBookmarkAction() -> SingleActionViewModel {
-        return isBookmarked ? getRemoveBookmarkAction() : getAddBookmarkAction()
+        return isBookmarked ? isBookmarkRefactorEnabled ? getEditBookmarkAction() : getRemoveBookmarkAction()
```

Comment:
Agreed, and I don't think this is a nit. Ternary operators that are not immediately obvious should not be in code. We can just do a simple if based return here for clarity.

Reply:
@DanielDervishi @adudenamedruby I try to avoid nested conditionals if possible. Would you consider this to be easier to read

```swift
private func getBookmarkAction() -> SingleActionViewModel {
    guard isBookmarked else { return getAddBookmarkAction() }
    return isBookmarkRefactorEnabled ? getEditBookmarkAction() : getRemoveBookmarkAction()
}
```

It takes out the nested ternary. Or do you think sticking with nested conditionals is optimal:

```swift
private func getBookmarkAction() -> SingleActionViewModel {
    if isBookmarked {
        if isBookmarkRefactorEnabled {
            return getEditBookmarkAction()
        }
        return getRemoveBookmarkAction()
    } else {
        return getAddBookmarkAction()
    }
}
```

Reply:
The first option looks better to me, alternatively you could do something like

```swift
private func getBookmarkAction() -> SingleActionViewModel {
    if isBookmarked {
        return isBookmarkRefactorEnabled ? getEditBookmarkAction() : getRemoveBookmarkAction()
    } else {
        return getAddBookmarkAction()
    }
}
```
but either this or the first looks alright to me :)

Reply:
Oh whoops just realized rudy commented not to include ternary operators in code. You could also do something like
```swift

private func getBookmarkAction() -> SingleActionViewModel {
  // Not bookmarked
  if !isBookmarked {
   return getAddBookmarkAction()
  // bookmarked and bookmark refactor is enabled
  } else if isBookmarkRefactorEnabled {
   return getEditBookmarkAction()
  // Bookmarked and bookmark refactor disabled
  } else  {
   return getRemoveBookmarkAction()
  }
}


```

Reply:
to clarify, I'm happy with ternary operators, just not stacked ones.
@MattLichtenstein your first option looks great to me.